probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -269,6 +279,36 @@ impl<S: ArmDebugState> ArmCommunicationInterface<S> {
 pub trait DapProbe: RawDapAccess + DebugProbe {}
 
 impl ArmProbeInterface for ArmCommunicationInterface<Initialized> {
+    /// If you're invoking this from a debug sequence, know that `reinitialize` will call back
+    /// onto you! Specifically, it will invoke some sequence of `debug_port_*` sequences with
+    /// varying internal state. If you're not prepared for this, you might recurse.
+    ///
+    /// `reinitialize` does handle `debug_core_start` to re-initialize any core's debugging.
+    /// If you're a chip-specific debug sequence, you're expected to handle this yourself.
```

Comment:
This should be a comment on the trait declaration, code using the trait must be aware of this.

Reply:
I put this on the actual implementation, because it wasn't clear that every implementation would call back into the parent object. I thought maybe it was just for this one particular instance.

I can move it to the trait documentation to note that it *may* call into the parent.

Reply:
True, if it's on the trait then it doesn't necessarily call back to the sequence.

But that would then mean that e.g. the debug_core_start sequence is not always used, which would be a bit problematic. Ideally this should behave the same, regardless of the probe.